src/caffe/proto/caffe.proto

```diff
@@ -597,6 +598,16 @@ message ConvolutionParameter {
   optional bool force_nd_im2col = 17 [default = false];
 }
 
+message CropParameter {  
+  // If only one crop_offset is specified, all the dimensions after the specified
+  // axis would be cropped (by the same amount); otherwise the number of crop
+  // values specified must be equal to the number of dimensions following axis, and
+  // the trailing dimensions would be cropped accordingly.
+  // Protip: standard dimensions are ( N,C,H,W )
+  optional uint32 axis = 1 [default = 2];
```

Comment:
Let's make `axis` mirror the `axis` arg of other layers to allow for negative indexing through an `int32` field and the `CanonicalAxisIndex()` helper method? See for instance https://github.com/BVLC/caffe/blob/master/src/caffe/layers/flatten_layer.cpp#L12-L13